src/Parallel/Algorithm.hpp

```diff
@@ -222,6 +222,32 @@ class AlgorithmImpl<ParallelComponent, tmpl::list<PhaseDepActionListsPack...>>
   template <typename Action>
   void simple_action() noexcept;
 
+  /// \brief Perform a blocking call the action `Action` with the current
+  /// \ref DataBoxGroup "DataBox", returning the return value of
+  /// `Action::apply`.
+  ///
+  /// \note `Action` must have a type alias `return_type` specifying its return
+  /// type. This constraint is to simplify the variant visitation logic for the
+  /// \ref DataBoxGroup "DataBox".
+  template <typename Action, typename... Args>
+  typename Action::return_type local_sync_action(Args... args) noexcept {
```

Comment:
What do you think about adding a function to `Parallel/Invoke.hpp` to wrap the call to this so that the template keyword isn't needed because calls like `->template local...`  are avoided?